e2e/test/clusterautoscaler/main_test.go

```diff
@@ -0,0 +1,64 @@
+// +build k8srequired
+
+package clusterautoscaler
+
+import (
+	"testing"
+
+	"github.com/giantswarm/microerror"
+
+	"github.com/giantswarm/azure-operator/v4/e2e/env"
+	"github.com/giantswarm/azure-operator/v4/e2e/setup"
+)
+
+var (
+	config     setup.Config
+	autoscaler *ClusterAutoscaler
+)
+
+func init() {
+	var err error
+	{
+		config, err = setup.NewConfig()
+		if err != nil {
+			panic(microerror.JSON(err))
+		}
+	}
+
+	var p *Provider
+	{
+		c := ProviderConfig{
+			AzureClient: config.AzureClient,
+			G8sClient:   config.Host.G8sClient(),
+			Logger:      config.Logger,
+			ClusterID:   env.ClusterID(),
+			CtrlClient:  config.K8sClients.CtrlClient(),
+		}
+
+		p, err = NewProvider(c)
+		if err != nil {
+			panic(microerror.JSON(err))
+		}
+	}
+
+	{
+		c := Config{
+			Logger:          config.Logger,
+			Guest:           config.Guest,
+			Provider:        p,
+			ClusterID:       env.ClusterID(),
```

Comment:
I believe you don't need `ClusterID` or `TargetNamespace`

Reply:
removed